### PR TITLE
TW-366 Reduce Frame rate when typing text in Chat

### PR DESCRIPTION
**Pull request by drminh2807**
# Issue
- #366 

# Before

https://github.com/linagora/twake-on-matrix/assets/12546908/dd7d84b8-91db-4cba-8e21-89e188e10d3c

# After

https://github.com/linagora/twake-on-matrix/assets/12546908/adc06921-131b-45d7-af1e-f5cdb1fdf63c

### CODE REVIEW

**Comment by sherlockvn** [comment]
> # Issue
> * [Reduce Frame rate when typing text in Chat #366](https://github.com/linagora/twake-on-matrix/issues/366)
> 
> # Before
> 
> # After
> 

can you add a video demo, thanks

**Comment by nqhhdev** [comment]
Pls add a demo in some cases: 
- Long mess
- Typing very fast
- send and type continuously

File: lib/pages/chat/chat_input_row.dart [hunk 1]
```diff
@@ -163,27 +163,44 @@ class ChatInputRow extends StatelessWidget {
...
+                                  valueListenable: controller.inputText,
+                                  builder: (context, value, child) {
+                                    return TwakeIconButton(
+                                      paddingAll: value.isEmpty ? 5.0 : 12,
+                                      tooltip: L10n.of(context)!.emojis,
+                                      onPressed: () => controller.emojiPickerAction(),
+                                      icon: Icons.tag_faces,
+                                    );
+                                  }
+                                )
```
Comment (dab246) [inline lib/pages/chat/chat_input_row.dart:176]: indent

File: lib/pages/chat/chat_input_row.dart [hunk 2]
```diff
@@ -163,27 +163,44 @@ class ChatInputRow extends StatelessWidget {
...
+                                  valueListenable: controller.inputText,
+                                  builder: (context, value, child) {
+                                    return TwakeIconButton(
+                                      paddingAll: value.isEmpty ? 5.0 : 12,
+                                      tooltip: L10n.of(context)!.keyboard,
+                                      onPressed: () => controller.inputFocus.requestFocus(),
+                                      icon: Icons.keyboard,
+                                    );
+                                  }
+                                ),
```
Comment (dab246) [inline lib/pages/chat/chat_input_row.dart:187]: same

**Comment by drminh2807** [comment]
@nqhhdev  @sherlockvn  Updated videos

File: lib/pages/chat/chat_input_row.dart (hunk 2 above)
Comment (drminh2807) [reply lib/pages/chat/chat_input_row.dart]: done

File: lib/pages/chat/chat_input_row.dart (hunk 1 above)
Comment (drminh2807) [reply lib/pages/chat/chat_input_row.dart]: done

File: lib/pages/chat/chat_input_row.dart [hunk 3]
```diff
@@ -163,27 +163,44 @@ class ChatInputRow extends StatelessWidget {
...
+                                      valueListenable: controller.inputText,
+                                      builder: (context, value, child) {
+                                        return TwakeIconButton(
+                                          paddingAll: value.isEmpty ? 5.0 : 12,
+                                          tooltip: L10n.of(context)!.keyboard,
+                                          onPressed: () => controller.inputFocus
+                                              .requestFocus(),
+                                          icon: Icons.keyboard,
+                                        );
+                                      }),
```
Comment (hoangdat) [inline lib/pages/chat/chat_input_row.dart:187]: Indent = 2 spaces

**Comment by hoangdat** [comment]
lgtm, but please verify the impacts for some kinds of message: emoji, files, images, link, tag, ...

**Comment by drminh2807** [comment]
> lgtm, but please verify the impacts for some kinds of message: emoji, files, images, link, tag, ...

@hoangdat  No impact

https://github.com/linagora/twake-on-matrix/assets/12546908/1017517f-723c-49ee-bce9-5faafc65d52b

**Review by hoangdat — APPROVED** [review]
Please rebase

**Comment by drminh2807** [comment]
> Please rebase

done

**Pull request merged**